crawl-ref/source/mon-util.h

```diff
@@ -290,8 +290,13 @@ mon_intel_type mons_intel(const monster& mon);
 // since the class variants do not handle zombies correctly.
 habitat_type mons_habitat_type(monster_type t, monster_type base_t,
                                bool real_amphibious = false);
+// The monster type should already be adjusted for zombie types
+habitat_type mons_habitat_type(monster_type t, bool real_amphibious = false);
```

Comment:
Functions in `mon-util.cc` whose first parameter is `monster_type` *mostly* start with the prefix `mons_class_`. For consistency it might be good to update this function name to `mon_class_habitat_type`.

That said, `mons_zombie_size`, `mons_skeleton`, and even the current `mons_habitat_type` right above this new function don't have that prefix and have been in the codebase for years... so maybe we hand-wave that away. 😅

Reply:
I've renamed the function to `mons_class_habitat` as it is just the class version of `mons_habitat`.